kedro/config/config.py

```diff
@@ -100,6 +102,15 @@ def __init__(
             "logging": ["logging*", "logging*/**", "**/logging*"],
         }
         self.config_patterns.update(config_patterns or {})
+        if conf_source.endswith(".zip"):
```

Comment:
That's an excellent point! Let me do some digging.

Reply:
So you can use: 

```python
import mimetypes
mimetypes.guess_type(my_file_path)
```

but this doesn't seem to work if your file doesn't have any extension at all. I'm quite surprised that the `tar -cvzf tar_conf conf` command works well for you. On my laptop that's not recognised as a compressed file..

Reply:
I tried with both the cli `tar -cvzf tar_conf conf` and with the `tarfile` package in python. There's doesn't seem to be a restriction or requirement on the file extension. (Not that it would happen commonly but you can even do `tar -cvzf tar_conf.zip conf`)
For tar files, we can check with the `tarfile` module with `tarfile.is_tarfile(filepath)`. Not sure if there is a similar thing for zip files.

Reply:
For zip files, using `zipfile` package if we initialise a `ZipFile` object with path to a file that's not a zip file, it throws a `BadZipFile` error (even if the extension is `.zip`).
Interestingly, if you do `zip -r zip_conf conf`, it creates the zip file with the `.zip` extension but you can create zip file without an extension if you create it using the `zipfile` python package similar to `tarfile`